src/actions/release-charm/helpers.ts

```diff
@@ -0,0 +1,35 @@
+import { Release } from '../../types';
+
+function getResourcesInfoByRelease(
+  releaseData: Release
+): Array<{ resourceName: string; resourceRev: string }> {
+  const body = releaseData.body
+    ?.split('\n')
+    .map((line) => line.trim())
+    .filter(
+      (line) => line.startsWith('-') || line.startsWith('resource-revision:')
+    );
+  if (!body) {
+    throw new Error(`Cannot find release body in tag ${releaseData.tag_name}`);
+  }
+  const resourceNames = body
+    .filter((line) => line.startsWith('-'))
+    .map((line) => line.split(':')[0].split(' ')[2]);
+  const resourceRevs = body
+    .filter((line) => line.startsWith('resource-revision:'))
+    .map((line) => line.split(':')[1].trim());
+
+  if (resourceNames.length === 0 || resourceRevs.length === 0) {
+    throw new Error(
+      `Cannot find resources info in release with tag ${releaseData.tag_name}`
+    );
+  }
+
+  const result = resourceNames.reduce((acc, field, index) => {
+    acc.push({ resourceName: field, resourceRev: resourceRevs[index] });
+    return acc;
+  }, [] as Array<{ resourceName: string; resourceRev: string }>);
+  return result;
+}
+
+export { getResourcesInfoByRelease };
```

Comment:
Since this action accepts either origin channel or revision number, release body is the only place with all the information. If I want to release rev5 to latest/beta, but rev5 is not the latest revision on any channels, I won't be able to find the combination of resources used.

Reply:
@agathanatasha, I think you're saying that we may need to promote something that is not currently in any channel (eg: promote a new revision to edge), `charmcraft status` only gives details of charms currently in a channel?

Reply:
That's correct. `charmcraft status` only show resources details for the latest revision in a channel.

Reply:
between this and the above comment, I think the discussion is really about scope.  Should this action handle cases of publishing to a channel, or just promoting from one channel to another.

Reply:
I believe that us having the flexibility of releasing any revision to a channel is going to be very important. For one it would allow us to rollback a release that had a breaking change. Additionally, we don't have to worry about our channels changing between the time we decide promoting something to a stable channel and us clicking the button.

Reply:
I really want to stress that I think @agathanatasha has done some great work here and demonstrated a lot of ingenuity in solving this.

> (eg: promote a new revision to edge)

That is already done automatically by the upload-charm action.

> I believe that us having the flexibility of releasing any revision to a channel is going to be very important. 

Parsing the content of a tag message that may or may not be there and using that to make decisions in our release automation just isn't within the realm of what I consider an acceptable risk. The value of that flexibility just isn't anywhere near outweighing the added fragility. 

> it would allow us to rollback a release that had a breaking change.

Promoting something to `stable` and then having to do a rollback is something I imagine happening very seldom, if ever. For all other channels, fixing forward is a perfectly acceptable workflow. In my opinion, the frequency of this practice does not warrant taking it into consideration.

> Additionally, we don't have to worry about our channels changing between the time we decide to promote something to a stable channel and us clicking the button.

If the risk for the content of a channel changing between you deciding to promote something and clicking the button is real, I don't think that will be solved by technology.

Reply:
> Parsing the content of a tag message that may or may not be there and using that to make decisions in our release automation just isn't within the realm of what I consider an acceptable risk. The value of that flexibility just isn't anywhere near outweighing the added fragility.

I completely agree that parsing the tag messages is not a good way of handling this. But us currently being limited in the way we can talk to charmhub and coming up with this suboptimal workaround  does not mean the initial use case is invalid. I still believe that being able to promote certain revisions is an important use case, specially considering how Field relies heavily on pinning revisions for any kind of customer deployments. 

To move this PR forward we can either leave it like it is or remove the revision specific promotion part and wait until charmcraft allows us to get more details from a certain charm revision even if it is not currently in a channel. @jnsgruk what are your thought on this?